micromagneticdata/tests/test_drive.py

```diff
@@ -80,3 +82,24 @@ def test_ovf2vtk(self):
     def test_slider(self):
         for drive in self.data:
             assert isinstance(drive.slider(), ipywidgets.IntSlider)
+
+    def test_to_xarray(self):
+        for drive in self.data:
+            assert isinstance(drive.to_xarray(), xr.DataArray)
+            assert all(
+                item in drive.to_xarray().attrs.items() for item in drive.info.items()
+            )
+            if drive.info["driver"] != "MinDriver":
+                assert np.allclose(
+                    drive.to_xarray()[drive.table.x].values,
+                    drive.table.data[drive.table.x].to_numpy(),
+                )
+
+            if drive.info["driver"] == "HysteresisDriver":
+                assert all(
+                    np.allclose(
+                        drive.to_xarray()[f"B{i}_hysteresis"].values,
+                        drive.table.data[f"B{i}_hysteresis"].to_numpy(),
+                    )
+                    for i in "xyz"
+                )
```

Comment:
Can we please add a check that the number of elements along the new (time/hysteresis/step) dimension is correct.